ds3-autogen-go/src/main/resources/tmpls/go/request/with_headers.ftl

```diff
@@ -0,0 +1,12 @@
+func (${name?uncap_first} *${name}) WithMetaData(key string, value string) *${name} {
+    if strings.HasPrefix(strings.ToLower(key), AMZ_META_HEADER) {
+        ${name?uncap_first}.headers.Add(key, value)
```

Comment:
No it does not lower case them. If I travel through several layers of interfaces I get the documentation
```
// CanonicalMIMEHeaderKey returns the canonical format of the
// MIME header key s. The canonicalization converts the first
// letter and any letter following a hyphen to upper case;
// the rest are converted to lowercase. For example, the
// canonical key for "accept-encoding" is "Accept-Encoding".
// MIME header keys are assumed to be ASCII only.
// If s contains a space or invalid header field bytes, it is
// returned without modifications.
```